app/jobs/geocode_location_job.rb

```diff
@@ -47,6 +51,6 @@ def enqueue_next_job
   end
 
   def next_location
-    Location.where('latitude IS NULL OR longitude IS NULL').first
+    Location.where(geocode_response: nil).first
```

Comment:
Gotcha, so `geocode_response` is never actually `nil` because it is explicitly set to that error string whenever there is no match at all from `Geocoder.search`. Ok that makes sense — thanks for clarifying the fallback logic.